src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1593,6 +1593,87 @@ public void MainWindowHandle_GetWithGui_ShouldRefresh_Windows()
             }
         }
 
+        [Fact]
+        [OuterLoop]
+        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Pops UI
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void MainWindowTitle_GetWithGui_ShouldRefresh_Windows()
+        {
+            const string ExePath = "notepad.exe";
+            Assert.True(IsProgramInstalled(ExePath));
+
+            using (Process process = Process.Start(ExePath))
+            {
+                try
+                {
+                    Assert.Equal(string.Empty, process.MainWindowTitle);
+
+                    for (int attempt = 0; attempt < 50; ++attempt)
+                    {
+                        process.Refresh();
+                        if (process.MainWindowTitle != string.Empty)
+                        {
+                            break;
+                        }
+
+                        Thread.Sleep(100);
+                    }
+
+                    Assert.NotEqual(string.Empty, process.MainWindowTitle);
+                }
+                finally
+                {
+                    process.Kill();
+                    Assert.True(process.WaitForExit(WaitInMS));
+                }
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Pops UI
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Responding_GetWithGui_ShouldRefresh_Windows()
+        {
+            const string ExePath = @"C:\Program Files\Windows NT\Accessories\wordpad.exe";
+            Assert.True(IsProgramInstalled(ExePath));
+
+            string dummyFilePath = $@"{Path.GetTempPath()}dummy_file";
+
+            var dummyFile = new FileStream(dummyFilePath, System.IO.FileMode.Create);
+            _ = dummyFile.Seek(2048L * 1024 * 1024, SeekOrigin.Begin);
+            dummyFile.WriteByte(0);
+            dummyFile.Close();
+
+            using (Process process = Process.Start(ExePath, dummyFilePath))
+            {
+                try
+                {
+                    Assert.True(process.Responding);
+
+                    for (int attempt = 0; attempt < 100; ++attempt)
+                    {
+                        process.Refresh();
+                        if (!process.Responding)
+                        {
+                            break;
+                        }
+
+                        Thread.Sleep(100);
+                    }
+
+                    Assert.False(process.Responding);
```

Comment:
Similar question; why do we expect Responding to be false after a Refresh?

Reply:
I see this behaviour with the test program that I've written:
https://github.com/JeroenOortwijn/Responding-test

With .Net Framework 4.8, the first `Process.Responding` call (immediately after Wordpad is started) returns true.
After a couple of seconds, Wordpad is so busy loading the document that the title bar shows "(Not Responding)". At that point, `Process.Responding` returns false.

Due to issue #36768, `Process.Responding` keeps returning true on .Net Core 3.1 and .Net 5.